coremltools/models/utils.py

```diff
@@ -206,16 +206,31 @@ def load_spec(filename):
     --------
     save_spec
     """
-    if _ModelPackage is None:
-        raise Exception(
-            "Unable to load libmodelpackage. Cannot make save spec."
-        )
+    name, ext = _os.path.splitext(filename)
+    
+    is_package = False
+
+    if not ext:
+        filename = "{}{}".format(filename, _MLMODEL_EXTENSION)
```

Comment:
I don't think we should be assuming it's an mlmodel here.

Reply:
That is a fair point. The logic here was duplicated from `save_spec`:
https://github.com/apple/coremltools/blob/main/coremltools/models/utils.py#L146

Given that this is the load case, it might be good for me to just take out the enforcement of extension existence in the filename, and also perhaps remove the exception generation at:
https://github.com/PierceLBrooks/coremltools/blob/fix-ml-model-load-spec-interpretation-as-ml-package/coremltools/models/utils.py#L220

Reply:
Ok, that's a really good point; let's be consistent with `save_spec`.